### PR TITLE
Clean code

### PR DESCRIPTION
**Pull request by joamatab**
make code more readable with f-strings

### CODE REVIEW

File: .github/workflows/pytest.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ jobs:
   pytest:
     strategy:
       matrix:
-        python-version: [3.5, 3.6, 3.7, 3.8, 3.9]
+        python-version: [3.6, 3.7, 3.8, 3.9, 3.10]
```
Comment (basnijholt) [inline .github/workflows/pytest.yml:12]: Perhaps do 
`[3.6, 3.7, 3.8, 3.9, "3.10"]`

**Comment by amccaugh** [comment]
Deprecated Python 3.5 and added 3.10, thanks @joamatab

**Comment by amccaugh** [comment]
@joamatab , how were these changes made?  With some kind of automated reformatter?

File: phidl/device_layout.py [hunk 2]
```diff
@@ -321,15 +321,13 @@ def _simplify(points, tolerance=0):
...
-        result1 = _simplify(M[: index + 1], tolerance)
-        result2 = _simplify(M[index:], tolerance)
+    if dmax <= tolerance:
+        return np.array([start, end])
 
-        result = np.vstack((result1[:-1], result2))
-    else:
-        result = np.array([start, end])
+    result1 = _simplify(M[: index + 1], tolerance)
+    result2 = _simplify(M[index:], tolerance)
```
Comment (basnijholt) [inline phidl/device_layout.py:329]: These changes seem unrelated to f-strings. Was this put here accidentally?

**Comment by basnijholt** [comment]
I am going to guess Sourcery.ai.

**Comment by joamatab** [comment]
yes https://docs.sourcery.ai/

went at Python conference last week and met the sourcery group

**Comment by amccaugh** [comment]
Hmm, I don't like several of these changes so I might cherry-pick them a little bit.  For instance, this is more compact but significantly less clear/obvious:

image


I'm also a little concerned that they're functionally the same -- I know it's supposed to be automated, but I'm not confident with how thoroughly tested it is

**Comment by joamatab** [comment]
Hi Adam,

i used to think the same way, now I find the right one line `if case` statement much easier to read :)

feel free to cherry pick

**Comment by basnijholt** [comment]
Just sharing my opinion (no one asked 😜) but I agree with @joamatab.

**Comment by amccaugh** [comment]
I've spent some time mulling this over and I think I'm not a fan of most of these changes.  It's generating a LOT of changes that don't seem meaningful, and at the same time and I don't trust that the automation is battle-tested enough that it's not going to screw up some edge case.  I looked at the sourcery.ai documentation, and they don't seem to have any proof (math/CS type proof) that what they're doing is functionally identical, they only run "existing tests on common packages" as a means of checking.  I like the f-string replacement, so if it's possible to submit a separate PR with just that I think I'd be on board.  Maybe in a few years when more projects are using this kind of thing I might be more open to it, but I really don't want to spend 20 hours tracking down some edge-case bug generated by these changes just for the sake of having the code look more "pythonic" (which I don't necessarily agree with in the first place)
